### PR TITLE
Add password support to Apple for recurring in-app purchases

### PR DESCRIPTION
**Pull request by chad3814**
- also add setup() and config() support to apple
- committed constants.js

### CODE REVIEW

File: lib/apple.js [hunk 1]
```diff
@@ -11,12 +11,53 @@ var errorMap = {
...
+        config = null;
+        return;
+    }
+};
+
+module.exports.setup = function (cb) {
+	if (!config || !config.applePassword) {
+
+		if (process.env.APPLE_IAP_PASSWORD) {
+            config = config || {};
```
Comment (voltrue2) [inline lib/apple.js]: Would you please fix the indentations here and the line below?

File: lib/google.js [hunk 2]
```diff
@@ -22,18 +26,31 @@ module.exports.reset = function () {
...
 
 module.exports.readConfig = function (configIn) {
-	config = configIn;
-	if (!config) {
+	if (!configIn) {
 		// no google iap or public key(s) from ENV variables
 		return;
 	}
+    config = {};
+    var config_value_set = false;
```
Comment (voltrue2) [inline lib/google.js]: I am being a little picky here :-)
Would  you change the variable name to configValueSet? I would like to use camel casing for all variable names.

Comment (chad3814) [inline lib/google.js]: sure no problem

File: lib/apple.js (hunk 1 above)
Comment (chad3814) [inline lib/apple.js]: sure, I didn't realize you were using tabs, my editor is setup for spaces

Comment (voltrue2) [inline lib/apple.js]: Great. Thanks.
Yeah. I usually use tabs in my editor.

Thanks for the PR by the way :-)

File: lib/apple.js [hunk 3]
```diff
@@ -11,12 +11,53 @@ var errorMap = {
...
+	return key.match(/^apple/);
+}
+
+module.exports.readConfig = function (configIn) {
+	if (!configIn) {
+		// no apple iap or password not required
+		return;
+	}
+	config = {};
+	var config_value_set = false;
```
Comment (voltrue2) [inline lib/apple.js]: Oops sorry, I missed this one last time..
Would you change  the variable name here as well?

File: lib/apple.js (hunk 1 above)
Comment (chad3814) [inline lib/apple.js]: np, we're using this at work for our app

Comment (voltrue2) [inline lib/apple.js]: Awesome! It's always nice  to hear that the library is being useful to others!

**Comment by voltrue2** [comment]
Thanks for the PR. Merged!

**Comment by chad3814** [comment]
oh don't suppose you'd bump the version number and do a publish?

**Comment by voltrue2** [comment]
Well I'm running the unit tests right now. If all tests are working I will probably be publishing a new version!On Thu, Mar 26, 2015 at 12:09 PM,  voltrue2/in-app-purchase <reply@reply.github.com> wrote:            oh don't suppose you'd bump the version number and do a publish?

**Pull request merged**